### PR TITLE
Add big-endian support in the patchmaker. Also add ARM BE8 support

### PR DESCRIPTION
**Pull request by paulnoalhyt**
- [x] I have reviewed the [OFRAK contributor guide](https://ofrak.com/docs/contributor-guide/getting-started.html) and attest that this pull request is in accordance with it.

**One sentence summary of this PR (This should go in the CHANGELOG!)**
Add support for compiling/assembling big-endian code with the patchmaker. Same for ARM BE8.

**Link to Related Issue(s)**

**Please describe the changes in your request.**
- Added big-endian flags in the GNU toochain for the compiler, linker and assembler.
- Added a `GENERIC_ARM_BE8` ProcessorType
- Added the corresponding BE8 flags to the compiler and linker

**Anyone you think should look at this, specifically?**
@whyitfor could maybe help adding tests to this.

### CODE REVIEW

**Comment by alchzh** [comment]
Two questions:
- Should we process the endian flag on an architecture by architecture basis instead of in abstract, since it is architecture dependent? I believe that PowerPC defaults to big endian and should add `-mlittle-endian` if little endian is specified? For other architectures, I think that we should probably raise an error if an unsupported endianness is used.
- Should the flag be added to llvm as well?

File: ofrak_patch_maker/ofrak_patch_maker/toolchain/gnu.py [hunk 1]
```diff
@@ -46,7 +46,6 @@ def __init__(
                 #      if sections contain more than one function =/
                 "-fno-merge-constants",  # avoids sections like .rodata.cst16, .rodata.str1.1 etc
                 "-fno-reorder-functions",
-                "-Wall",
```
Comment (alchzh) [inline ofrak_patch_maker/ofrak_patch_maker/toolchain/gnu.py:49]: What is the reason for this change? I don't see any additional warnings generated by the big endian test cases you added. If there's a specific spurious warning you ran into consistently, it should be disabled by a conditional `-Wno` in `gnu_arm.py`

**Review by alchzh — DISMISSED** [review]

File: ofrak_patch_maker/ofrak_patch_maker/toolchain/gnu.py (hunk 1 above)
Comment (paulnoalhyt) [reply ofrak_patch_maker/ofrak_patch_maker/toolchain/gnu.py:49]: I just removed one because the `-Wall` option was present twice in this list

**Review by alchzh — APPROVED** [review]
